pkg/engine/rule/r0002_unexpected_file_access.go

```diff
@@ -58,6 +75,32 @@ func (rule *R0002UnexpectedFileAccess) ProcessEvent(eventType tracing.EventType,
 		return nil
 	}
 
+	if rule.shouldIgnoreMounts {
+		rule.mutex.RLock()
+		mounts, ok := rule.containerIdToMountPaths[openEvent.ContainerID]
+		rule.mutex.RUnlock()
+		if !ok {
+			err := rule.setMountPaths(openEvent.PodName, openEvent.Namespace, openEvent.ContainerID, openEvent.ContainerName, engineAccess)
+			if err != nil {
+				log.Printf("Failed to set mount paths: %v", err)
+				return nil
+			}
+			rule.mutex.RLock()
+			mounts = rule.containerIdToMountPaths[openEvent.ContainerID]
+			rule.mutex.RUnlock()
+		}
+
+		for _, mount := range mounts {
+			contained := isPathContained(mount, openEvent.PathName)
+			if contained {
+				if os.Getenv("DEBUG") == "true" {
+					log.Printf("Path %s is mounted in pod %s/%s - Skipping check", openEvent.PathName, openEvent.Namespace, openEvent.PodName)
+				}
+				return nil
+			}
+		}
+	}
+
```

Comment:
This is not a blocking comment, but you can "set" all the mount paths (by container name) at the creation time of the rule then you don't need mutex at all :)

Also note, that rule instance is per container, therefore we will have only one item in the ` rule.containerIdToMountPaths` slice

Reply:
Not really because we do not pass the needed details in the rule creation.